pkg/webhook/topic_validator.go

```diff
@@ -121,9 +145,65 @@ func (s *webhookServer) validateKafkaTopic(topic *banzaicloudv1alpha1.KafkaTopic
 		log.Info(fmt.Sprintf("Spec is requesting replication factor of %v, larger than cluster size of %v", topic.Spec.ReplicationFactor, broker.NumBrokers()))
 		return notAllowed(invalidReplicationFactorErrMsg, metav1.StatusReasonBadRequest)
 	}
+	return nil
+}
 
-	// everything looks a-okay
-	return &admissionv1beta1.AdmissionResponse{
-		Allowed: true,
+// checkExistingKafkaTopicCRs checks whether there's any other duplicate KafkaTopic CR exists
+// that refers to the same KafkaCluster's same topic
+func (s *webhookServer) checkExistingKafkaTopicCRs(ctx context.Context,
+	clusterNamespace string, topic *banzaicloudv1alpha1.KafkaTopic) *admissionv1beta1.AdmissionResponse {
+	// check KafkaTopic in the referred KafkaCluster's namespace
+	kafkaTopicList := banzaicloudv1alpha1.KafkaTopicList{}
+	err := s.client.List(ctx, &kafkaTopicList,
+		client.InNamespace(clusterNamespace),
+		client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector("spec.name", topic.Spec.Name),
+		},
+		client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector("spec.clusterRef.name", topic.Spec.ClusterRef.Name),
+		},
+	)
+	if err != nil {
+		log.Info("failed to list KafkaTopics")
+		return notAllowed("API failure while retrieving KafkaTopic list, please try again", metav1.StatusReasonServiceUnavailable)
+	}
+	var foundKafkaTopic *banzaicloudv1alpha1.KafkaTopic
+	for i, kafkaTopic := range kafkaTopicList.Items {
+		referredNamespace := kafkaTopic.Spec.ClusterRef.Namespace
+		if referredNamespace == "" || referredNamespace == clusterNamespace {
+			foundKafkaTopic = &kafkaTopicList.Items[i]
+		}
+	}
+	if foundKafkaTopic != nil {
+		msg := fmt.Sprintf("KafkaTopic CR '%s' in namesapce '%s' is already referencing Kafka topic '%s'",
+			foundKafkaTopic.Name, foundKafkaTopic.Namespace, foundKafkaTopic.Spec.Name)
+		log.Info(msg)
+		return notAllowed(msg, metav1.StatusReasonInvalid)
 	}
+
+	// check KafkaTopic in every namespace
+	kafkaTopicList = banzaicloudv1alpha1.KafkaTopicList{}
+	err = s.client.List(ctx, &kafkaTopicList,
+		client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector("spec.name", topic.Spec.Name),
+		},
+		client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector("spec.clusterRef.name", topic.Spec.ClusterRef.Name),
+		},
+		client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector("spec.clusterRef.namespace", topic.Spec.ClusterRef.Namespace),
+		},
+	)
+	if err != nil {
+		log.Info("failed to list KafkaTopics")
+		return notAllowed("API failure while retrieving KafkaTopic list, please try again", metav1.StatusReasonServiceUnavailable)
+	}
+	if len(kafkaTopicList.Items) > 0 {
```

Comment:
The list would contain the current `topic` so I think the check should rather be `if len(kafkaTopicList.Items) > 1 {`